### DISCUSSION TITLE
how properly build test deps?

### DISCUSSION THREAD

**Discussion by dzmitry-lahoda** [ipetkov/crane#141] [Q&A] [answered]
I tried to build deps only with extra arg `--tests` and then "build" tests with cargo command `cargo test` and relevant deps. As I see even if deps are cached before, test command still rebuilds deps. Same happens without `--tests` extra arg.

So question, what is proper way to build and share test deps?

**Comment by dpc** [comment, marked as answer]
Please link the code you have.

**Reply by dzmitry-lahoda** [reply]
https://github.com/ComposableFi/composable/blob/4e91eb6b68e69acc604512c04be37ea1140d1e4d/flake.nix#L841

**Reply by dpc** [reply]
I can't spend much time on it RN, but it seems you're doing a lot of things similar what I have: https://github.com/fedimint/fedimint/blob/master/flake.nix , so maybe something there will give you an idea.

**Reply by ipetkov** [reply]
@dzmitry-lahoda you have [`doCheck = false;`](https://github.com/ComposableFi/composable/blob/4e91eb6b68e69acc604512c04be37ea1140d1e4d/flake.nix#L228) set it to `true` and `buildDepsOnly` will cache test artifacts as well

**Reply by ipetkov** [reply]
Running `nix flake lock --update-inputs crane` also seems to help too

**Reply by dzmitry-lahoda** [reply]
setting doCheck = true and having check command as it was did not help. updating crane started to fail our builds weird ways. need  to fix that later.
